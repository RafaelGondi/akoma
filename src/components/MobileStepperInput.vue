<template>
	<div class="mobile__container">
		<input
			id="mobile-stepper-input"
			v-model="internalValue"
			:class="computedStepperClass"
			placeholder="0"
			type="number"
			:disabled="disabled"
		>
		<icon-button
			icon="minus-outline"
			size="lg"
			:disabled="disabled"
			@cds-click="$emit('minus')"
		/>
		<icon-button
			icon="plus-outline"
			size="lg"
			:disabled="disabled"
			@cds-click="$emit('plus')"
		/>
	</div>
</template>

<script>
import IconButton from './IconButton.vue';
import stateValidator from '../utils/validators/state';

export default {
	components: {
		IconButton,
	},

	props: {
		modelValue: {
			type: Number,
			default: 0,
		},
		disabled: {
			type: Boolean,
			default: false,
		},
		state: {
			type: String,
			default: 'default',
			validator: stateValidator,
		}
	},

	data() {
		return {
			internalValue: 0,
		};
	},

	computed: {
		computedStepperClass() {
			if (this.state === 'default') {
				return 'mobile__stepper';
			}

			return `mobile__stepper--${this.state}`;
		},
	},

	watch: {
		modelValue(newValue) {
			this.internalValue = newValue;
		},

		internalValue(newValue) {
			if (newValue === this.modelValue) {
				return;
			}

			this.$emit('update:modelValue', newValue);
		}
	},
}
</script>

<style lang="scss" scoped>
@import '../assets/sass/tokens.scss';

.mobile {
	&__container {
		display: flex;
		align-items: center;
		gap: spacer(2);
	}

	&__stepper {
		border: 1px solid $n-50;
		border-radius: $border-radius-extra-small;
		height: 42px;
		width: 70px;	
		padding: pYX(4, 3);
		color: $n-600;
		margin: mr(1);

		&:focus {
			outline: 0;
		}

		&--valid {
			@extend .mobile__stepper;
			outline: 1px solid $gp-500;
		}

		&--invalid {
			@extend .mobile__stepper;
			outline: 1px solid $rc-600;
		}
	}
}
</style>

