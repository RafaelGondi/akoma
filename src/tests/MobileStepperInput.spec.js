import { describe, test, expect } from 'vitest';
import MobileStepperInput from '../components/MobileStepperInput.vue';
import { shallowMount } from '@vue/test-utils';

describe('MobileStepperInput', () => {
	test('renders correctly', async () => {
		const wrapper = shallowMount(MobileStepperInput);

		expect(wrapper.html()).toMatchSnapshot();
	});

	test('state classes are applyied correctly', async () => {
		expect.assertions(2);

		const wrapper = shallowMount(MobileStepperInput);

		expect(wrapper.vm.computedStepperClass).toBe('mobile__stepper');

		await wrapper.setProps({
			state: 'invalid',
		});

		expect(wrapper.vm.computedStepperClass).toBe('mobile__stepper--invalid');
	});
});