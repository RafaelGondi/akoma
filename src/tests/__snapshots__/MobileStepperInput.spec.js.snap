// Vitest Snapshot v1

exports[`MobileStepperInput > renders correctly 1`] = `
"<div data-v-962d09dd=\\"\\" class=\\"mobile__container\\"><input data-v-962d09dd=\\"\\" id=\\"mobile-stepper-input\\" class=\\"mobile__stepper\\" placeholder=\\"0\\" type=\\"number\\">
  <icon-button-stub data-v-962d09dd=\\"\\" size=\\"lg\\" icon=\\"minus-outline\\" disabled=\\"false\\" feedbackonclick=\\"false\\" feedbackicon=\\"check-outline\\" variant=\\"white\\"></icon-button-stub>
  <icon-button-stub data-v-962d09dd=\\"\\" size=\\"lg\\" icon=\\"plus-outline\\" disabled=\\"false\\" feedbackonclick=\\"false\\" feedbackicon=\\"check-outline\\" variant=\\"white\\"></icon-button-stub>
</div>"
`;
